dependencies.yaml

```diff
@@ -100,17 +100,25 @@ dependencies:
       - output_types: conda
         matrices:
           - matrix:
-              cuda: "11.2"
+              cuda: "11.5"
             packages:
-              - cudatoolkit=11.2
+              - cudatoolkit=11.5
+              - libcusparse-dev>=11.7.0.31,<=11.7.0.107
+              - libcusparse>=11.7.0.31,<=11.7.0.107
```

Comment:
Do we need both `libcusparse` and `libcusparse-dev`? Shouldn't `libcusparse-dev` include the runtime lib from `libcusparse` (or at least depend on it)?

Also what's up w/ the version? Is this versioned independently from `cudatoolkit`?

Reply:
Great questions! I struggled with both of these. I'll answer these in reverse order.

First, you are correct that the versions of individual CUDA Toolkit packages like `libcusparse[-dev]` are independent of the CUDA Toolkit version. In some GitHub Actions PRs, we have added comments about what CUDA Toolkit versions we're aiming for when pinning individual components -- I haven't checked that on this PR yet, we may need to add comments explaining that for cuSpatial if they aren't there yet.

Second, `libcusparse-dev` depends on `libcusparse`. We must pin both `libcusparse-dev` and `libcusparse` because the pinning is `>=`, meaning that pinning only `libcusparse-dev` to the versions that correspond to CUDA 11.5 means that the system could pull `libcusparse` runtime libraries for CUDA 12.0. Crossing CTK major versions is not a good thing, and we need to pin these in `dependencies.yaml` so that we are actually testing CTK components that we claim (e.g. CI tests for 11.2, 11.4, 11.5, ...)

For an example, see https://anaconda.org/nvidia/libcusparse-dev/files `linux-64/libcusparse-dev-11.7.0.107-h338262b_0.tar.bz2`.
![image](https://user-images.githubusercontent.com/3943761/209870314-dc3a3944-7827-40ec-ae69-15d4b61e833a.png)